nengo/spa/bind.py

```diff
@@ -0,0 +1,42 @@
+import nengo
+from nengo.spa.module import Module
+
+
+class Bind(Module):
+    """A module for binding together both inputs
+
+    Parameters
+    ----------
+    dimensions : int
+        Number of dimensions for the two vectors to be compared
+    vocab : Vocabulary, optional
+        The vocabulary to use to interpret the vectors
+    n_neurons : int
+        Number of neurons to use in each product computation
+    invert_a, invert_b : bool
+        Whether to reverse the order of elements in either
+        the first input (`invert_a`) or the second input (`invert_b`).
+        Flipping the second input will make the network perform circular
+        correlation instead of circular convolution.
+    input_magnitude : float
+        Effective input magnitude for the multiplication.
+        The actual input magnitude will be this value times sqrt(2)
+    """
+    def __init__(self, dimensions, vocab=None, n_neurons=200, invert_a=False,
+                 invert_b=False, input_magnitude=1.0, label=None, seed=None,
+                 add_to_container=None):
+        super(Bind, self).__init__(label, seed, add_to_container)
+        if vocab is None:
+            # use the default vocab for this number of dimensions
+            vocab = dimensions
```

Comment:
If you pass in a vocabulary that doesn't match the dimensions you're using the code throws an error about a bad connection. This isn't ideal, but since this issue doesn't just affect `bind.py`, but all of the spa modules, I moved it to another issue.